src/sssom/rdf_internal.py

```diff
@@ -0,0 +1,992 @@
+            if obj.datatype is None or obj.datatype == XSD.string:
+                if obj.value not in self.allowed_values:
+                    raise ValueError(f"Invalid enum value {obj}")
+                return obj.value
+
+        raise ValueError("Invalid node type (IRI or string literal expected)")
+
+    @override
+    def to_rdf(self, value: Any) -> Node:
+        """Convert a SSSOM enum value into a RDF node."""
+        # Make sure we have a _text_ value, regardless what the rest of
+        # SSSOM-Py can give us...
+        value = str(value)
+        uri = self.uris_by_value.get(value)
+        if uri is not None:
+            return uri
+        else:
+            # We do _not_ check whether the value is a valid value for
+            # the enum. If it is not, then something wrong happened
+            # upstream of us (a set was constructed or accepted with
+            # invalid values) and it's not up to us to deal with the
+            # the fallout.
+            return Literal(value)
+
+
+class ValueConverterFactory(object):
+    """Helper object to create value converters."""
+
+    constructors: Dict[str, Type[ValueConverter]]
+
+    def __init__(self) -> None:
+        """Create a new instance."""
+        self.constructors = {
+            "date": DateValueConverter,
+            "double": DoubleValueConverter,
+            "string": StringValueConverter,
+            "ncname": StringValueConverter,
+            "EntityReference": EntityReferenceValueConverter,
+            "uriorcurie": EntityReferenceValueConverter,
+            "NonRelativeURI": NonRelativeURIValueConverter,
+        }
+
+    def create(
+        self, range_name: str, schema: SchemaView, ccp: CurieConverterProvider
+    ) -> Optional[ValueConverter]:
+        """Create a new value converter.
+
+        :param range_name: The range for which a value converter is
+            wanted.
+        :param schema: The SSSOM LinkML schema.
+        :param ccp: The object that will provide the CURIE converter to
+            use, for the value converters that need one.
+
+        :returns: A suitable value converter for the range. May be None
+            if the range is not a scalar range.
+        """
+        if schema.get_class(range_name) is not None:
+            # This range is for objects, not scalar values
+            return None
+
+        if range_name.endswith("_enum"):
+            return EnumValueConverter(schema, range_name)
+
+        ctor = self.constructors.get(range_name)
+        if ctor == EntityReferenceValueConverter:
+            # CURIE provider needed
+            return EntityReferenceValueConverter(ccp)
+        elif ctor is not None:
+            return ctor()
+        else:
+            # This should only happen if a brand new type of slot has
+            # been introduced in the SSSOM schema
+            raise NotImplementedError(f"Range {range_name} is not supported")
+
+
+class ObjectConverter(object):
```

Comment:
inheriting from `object` isn't needed since Python 2

```suggestion
class ObjectConverter:
```